app/testOnly/models/formsmodel/StartJourneyForm.scala

```diff
@@ -18,38 +18,41 @@ package testOnly.models.formsmodel
 
 import essttp.journey.model.{Origin, Origins}
 import essttp.rootmodel.epaye.{TaxOfficeNumber, TaxOfficeReference}
-import essttp.rootmodel.{AmountInPence, EmpRef}
-import models.Language
+import essttp.rootmodel.{AmountInPence, EmpRef, Vrn}
 import models.MoneyUtil.amountOfMoneyFormatter
-import models.{EligibilityError, EligibilityErrors}
+import models.{EligibilityError, EligibilityErrors, Language}
 import play.api.data.Forms.{mapping, optional, seq}
-import play.api.data.{FieldMapping, Form, Forms, Mapping}
+import play.api.data._
 import testOnly.messages.Messages
 import testOnly.models.testusermodel.RandomDataGenerator
 import util.EnumFormatter
 
 import scala.util.Random
 
 final case class StartJourneyForm(
-    signInAs:          SignInAs,
-    enrolments:        Seq[Enrolment],
-    origin:            Origin,
-    eligibilityErrors: Seq[EligibilityError],
-    debtTotalAmount:   BigDecimal,
-    interestAmount:    Option[BigDecimal],
-    taxReference:      Option[String]
+    signInAs:            SignInAs,
+    enrolments:          Seq[Enrolment],
+    origin:              Origin,
+    eligibilityErrors:   Seq[EligibilityError],
+    payeDebtTotalAmount: Option[BigDecimal],
+    vatDebtTotalAmount:  Option[BigDecimal],
```

Comment:
instead of two `Option[BigDecimal]` for `payeDebtTotalAmount` and `vatDebtTotalAmount` it could be cleaner to have a separate `taxRegime: String` field which is a hidden field on each tab but can be used in the mapping to tell which tax regime we're in. Then you could have one non-optional `debtTotalAmount: BigDecimal` and perhaps even just one `taxReference: Option[String]`. What do you think?

Reply:
That does make sense, but the trouble I had was trying to obtain the input tax ref and using that to drive the validation on the amount field, have you got any examples of doing this?

Reply:
I think part of my issue is I also tried to make the amount without interest field optional (so it just defaults to £1234.53) for ease of use to save time getting into the journey